### DISCUSSION TITLE
DI for IFirebasePushNotification in Blazor

### DISCUSSION THREAD

**Discussion by konrad79** [thomasgalliker/Plugin.FirebasePushNotifications#35] [Q&A]
I've a Blazor Hybrid App - and want to use the plugin.
But how is this done with the dependency injection?

I inject the 
`@inject IFirebasePushNotification firebasePushNotification`
in my Index.razor. 
But do I have to do more than the 
`.UseFirebasePushNotifications()`
in the MauiProgram?

The code builds but on startup I get (of course) the error:

_"Cannot provide a value for property 'firebasePushNotification' on type 'BalzorHybridFirebaseTest2.Pages.Index'. There is no registered service of type 'Plugin.FirebasePushNotifications.IFirebasePushNotification'._

**Comment by thomasgalliker** [comment]
Unfortunately, I have never used blazer. UseFirebasePushNotifications not only registers required services, it also configures them. So maybe you habe to do something similar to what is done in UseFirebasePushNotifications method.

**Reply by konrad79** [reply]
Thanks for the fast response.

I struggled several times about functions and properties not named exactly as in the documentation ... e.g.:

```
.UseFirebasePushNotifications(
o =>
{
    o.AutoInit = true;
})
```

instead of `AutoInitEnabled`
Where there major changes in the past? Using nuget version 1.0.26 of Plugin.FirebasePushNotifications

And even if I got the DI to work, when I call `RegisterForPushNotifications()` I get the error 

`Default FirebaseApp is not initialized in this process com.companyname.balzorhybridfirebasetest2. Make sure to call FirebaseApp.initializeApp(Context) first
`
which perhaps indicates that the `AutoInit `is not doing what it should? The `google-services.json` is in the path `Platforms\Android\Resources` as in the documentation, if that is important.

**Reply by thomasgalliker** [reply]
Version 1.x of the nuget was a clone of the old, unmaintained plugin. 2.x is a complete refactoring and it is not yet finished. There was/is a lot of static, untestable code in the old 1.x code base. The documentation you found is probably from the old/unmaintained plugin — I have nothing to do with this old plugin and the latest code (develop branch) does no longer correspond to it.

My goal was to create a .NET MAUI Android/iOS plugin for Firebase push notification for one of my apps. Everything else is optional. If you need enhancements, bug fixes or support, its all based on best effort. I can also offer commercial support, if you need help immediately 👍🏻
